### PR TITLE
[Woo POS] Crash parcelizing lambda

### LINKED ISSUE #13146: [Woo POS] Crash when connecting reader, after backgrounding the app, and restoring state
Sentry: https://a8c.sentry.io/issues/6138913717/?project=1459556
Discovery: https://wp.me/pdfdoF-5ZI#comment-7145

Description:

In Woo POS a crash occurs when connecting reader. 

Steps to repro
1. Go to Woo POS (More > Point Of Sale)
2. Add items to cart
3. > Checkout
4. Click "Connect to reader"
5. Wait for reader connection dialog
6. Move app to background
7. Observe crash

Logs:
```kotlin
java.lang.ClassCastException: com.woocommerce.android.ui.woopos.home.totals.WooPosTotalsViewModel$$ExternalSyntheticLambda0 cannot be cast to java.io.Serializable
                                                                                                    	at com.woocommerce.android.ui.woopos.home.totals.WooPosTotalsViewState$ReaderStatus$Disconnected.writeToParcel(Unknown Source:22)
```

### PR DESCRIPTION
**Pull request by samiuelson**
Fix for crash: 
```kotlin
java.lang.ClassCastException
com.woocommerce.android.ui.woopos.home.totals.WooPosTotalsViewState$ReaderStatus$Disconnected.writeToParcel
```

Closes: #13146

### Description

This PR fixes the above crash by removing lambda from the state model, and handling button click by passing event to `VM::onUIEvent` method.

### Steps to reproduce

1. Open Point of sale (More > Point of sale)
2. Ensure reader is disconnected
3. Add item to cart and click "Checkout"
4. Wait for totals table to show up
5. Move app to background and observe crash

### Testing information

The above scenario must not result in crashing.

### The tests that have been performed

I tested that the crash is not happening, and the "Connect reader" button works.

### Images/gif

–

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewState.kt [hunk 1]
```diff
@@ -56,8 +55,7 @@ sealed class WooPosTotalsViewState : Parcelable {
         data class Disconnected(
             override val title: String,
             override val subtitle: String,
-            val actionButonLabel: String,
-            val onAction: () -> Unit,
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewState.kt:60]: 💡 Solving the parcelization crash by removing lambda from model.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt [hunk 2]
```diff
@@ -149,6 +149,7 @@ class WooPosTotalsViewModel @Inject constructor(
         cardReaderPaymentController?.stop()
     }
 
+    @Suppress("LongMethod")
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt:152]: np:

We may extract some events handling into separate methods

**Review by kidinov — APPROVED** [review]
LGTM! Crash is fixed.

I left one np, please take a look

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt (hunk 2 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt:152]: Good point. Let's do that in a separate PR.

**Pull request merged**